src/Pos/Crypto/HDDiscovery.hs

```diff
@@ -15,16 +16,28 @@ import           Pos.Txp.DB          (UtxoIter, runUtxoMapIterator)
 import           Pos.Util.Iterator   (MonadIterator (..))
 
 discoverHDAddress :: MonadDB m => HDPassphrase -> m [(Address, [Word32])]
-discoverHDAddress walletPassphrase =
-    runUtxoMapIterator @UtxoIter (step []) (txOutAddress . toaOut . snd)
+discoverHDAddress walletPassphrase = safeHead <$> discoverHDAddresses [walletPassphrase]
+  where
+    safeHead (x:[]) = x
+    safeHead _ = []
+
+discoverHDAddresses :: MonadDB m => [HDPassphrase] -> m [[(Address, [Word32])]]
```

Comment:
Can you write at least something about these functions (haddock comments)? It's not clear what they do.